src/TypeScript/Analyzer/JavaScriptAnalyzer.cs

```diff
@@ -55,33 +55,33 @@ internal sealed class JavaScriptAnalyzer : IAnalyzer, IDisposable
         [ImportingConstructor]
         public JavaScriptAnalyzer(IEslintBridgeClientFactory eslintBridgeClientFactory,
             IRulesProviderFactory rulesProviderFactory,
-            IActiveJavaScriptRulesProvider activeRulesProvider,
             ITelemetryManager telemetryManager,
             IAnalysisStatusNotifier analysisStatusNotifier,
             IActiveSolutionTracker activeSolutionTracker,
             IAnalysisConfigMonitor analysisConfigMonitor,
             ILogger logger)
-            : this(eslintBridgeClientFactory, activeRulesProvider,
-                new EslintBridgeIssueConverter(rulesProviderFactory.Create("javascript")),
-                telemetryManager,
-                analysisStatusNotifier,
-                activeSolutionTracker,
-                analysisConfigMonitor,
-                logger)
+            : this(eslintBridgeClientFactory,
+                  rulesProviderFactory.Create("javascript"),
+                  telemetryManager,
+                  analysisStatusNotifier,
+                  activeSolutionTracker,
+                  analysisConfigMonitor,
+                  logger)
         {
         }
 
-        internal JavaScriptAnalyzer(IEslintBridgeClientFactory eslintBridgeClientFactory,
-            IActiveJavaScriptRulesProvider activeRulesProvider,
-            IEslintBridgeIssueConverter issuesConverter,
+        internal /* for testing */ JavaScriptAnalyzer(IEslintBridgeClientFactory eslintBridgeClientFactory,
+            IRulesProvider rulesProvider,
             ITelemetryManager telemetryManager,
             IAnalysisStatusNotifier analysisStatusNotifier,
             IActiveSolutionTracker activeSolutionTracker,
             IAnalysisConfigMonitor analysisConfigMonitor,
-            ILogger logger)
+            ILogger logger,
+            IEslintBridgeIssueConverter issuesConverter = null // settable for testing
+            )
         {
-            this.activeRulesProvider = activeRulesProvider;
-            this.issuesConverter = issuesConverter;
+            this.rulesProvider = rulesProvider;
+            this.issuesConverter = issuesConverter ?? new EslintBridgeIssueConverter(rulesProvider);
```

Comment:
I changed the constructor pattern slightly so that we can still inject an issue converter for testing if we want to, but if we don't we'll get a default one.

This allows us to only call the factory once and store the returned `IRuleProvider` instance as well as using it to create the constructor.

There are other ways this could be done, but they involve either calling the factory twice, or creating an additional private constructor.